### PR TITLE
Add io.github.narunlifescience.AlphaPlot

### PR DESCRIPTION
**Pull request by filipestevao**
AlphaPlot is a free application for Scientific Data Analysis and Visualization.
https://github.com/narunlifescience/AlphaPlot

### CODE REVIEW

**Comment by filipestevao** [comment]
bot, build io.github.narunlifescience.AlphaPlot

**Comment by filipestevao** [comment]
io.qt.qtdatavis3d.BaseApp is a dependency

File: io.github.narunlifescience.AlphaPlot.metainfo.xml [hunk 1]
```diff
@@ -0,0 +1,33 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (bilelmoussaoui) [inline io.github.narunlifescience.AlphaPlot.metainfo.xml:1]: Should be upstreamed if possible.

Comment (filipestevao) [reply io.github.narunlifescience.AlphaPlot.metainfo.xml:1]: Just for the record, I'm not the upstream of this project, but he knows about this PR and [agreed to package AlphaPlot in flatpak](https://github.com/narunlifescience/AlphaPlot/issues/20#issue).

I'm going to talk to the upstream to see if it accepts adding the appdata in the source code.

File: io.github.narunlifescience.AlphaPlot.yml [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+command: alphaplot
+rename-icon: alphaplot
+rename-desktop-file: alphaplot.desktop
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --device=dri
+  - --filesystem=home
+  - --share=network
+  # Enable Wayland but keep X11 default for now
```
Comment (barthalion) [inline io.github.narunlifescience.AlphaPlot.yml:16]: I don't see the point. If X is the only supported "platform", just don't enable Wayland at all.

File: io.github.narunlifescience.AlphaPlot.yml [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  - --socket=wayland
+  - --env=QT_QPA_PLATFORM=xcb
+cleanup-commands:
+  - /app/cleanup-BaseApp.sh
+modules:
+- name: gsl
```
Comment (barthalion) [inline io.github.narunlifescience.AlphaPlot.yml:22]: You probably want to clean up `/include` and `*.a` too.

File: io.github.narunlifescience.AlphaPlot.metainfo.xml (hunk 1 above)
Comment (filipestevao) [reply io.github.narunlifescience.AlphaPlot.metainfo.xml:1]: It was upstreamed ([commit](https://github.com/narunlifescience/AlphaPlot/commit/70c42625fbe3a3ca81875b27d1ca93efb2955d9d))

**Comment by filipestevao** [comment]
bot, build io.github.narunlifescience.AlphaPlot

**Comment by filipestevao** [comment]
I made several changes. Please check that everything is fine.

File: patches/metainfo.patch [hunk 4]
```diff
@@ -0,0 +1,39 @@
+diff --git a/data/metainfo.xml b/data/metainfo.xml
+new file mode 100644
+index 00000000..6f37c6fe
+--- /dev/null
++++ b/data/metainfo.xml
```
Comment (bilelmoussaoui) [inline patches/metainfo.patch:5]: Would be nice to upstream this

File: io.github.narunlifescience.AlphaPlot.yml (hunk 3 above)
Comment (filipestevao) [reply io.github.narunlifescience.AlphaPlot.yml]: Updated "cleanup"

File: io.github.narunlifescience.AlphaPlot.yml (hunk 2 above)
Comment (filipestevao) [reply io.github.narunlifescience.AlphaPlot.yml]: Wayland support removed.

File: patches/metainfo.patch (hunk 4 above)
Comment (filipestevao) [reply patches/metainfo.patch:5]: metainfo.xml was recently upstreamed ([commit](https://github.com/narunlifescience/AlphaPlot/commit/70c42625fbe3a3ca81875b27d1ca93efb2955d9d)).

In the next release, this patch will be deleted.

**Comment by bilelmoussaoui** [comment]
Looks good to me, anyone else from upstream that would like to have write access to the repository once created?

**Comment by filipestevao** [comment]
Yes, you can add @narunlifescience

File: io.github.narunlifescience.AlphaPlot.yml [hunk 5]
```diff
@@ -0,0 +1,78 @@
...
+rename-desktop-file: alphaplot.desktop
+finish-args:
+  - --share=ipc
+  - --socket=x11
+  - --device=dri
+  - --env=SESSION_MANAGER= # required to avoid Qt warning
+  - --filesystem=xdg-desktop
+  - --filesystem=xdg-documents
+  - --filesystem=xdg-pictures
+  - --filesystem=xdg-public-share
```
Comment (bilelmoussaoui) [inline io.github.narunlifescience.AlphaPlot.yml:16]: Any reason for those? Not sure if the app uses QFileDialog but that uses the portal.

File: io.github.narunlifescience.AlphaPlot.yml [hunk 6]
```diff
@@ -0,0 +1,78 @@
...
+    buildsystem: qmake
+    build-options:
+      env:
+        - QMAKEPATH=/app/lib
+    config-opts:
+      - QMAKE_INCDIR+=/app/include/QtDataVisualization
+    sources:
+      - type: git
+        url: https://github.com/narunlifescience/AlphaPlot.git
+        tag: 1.011
```
Comment (bilelmoussaoui) [inline io.github.narunlifescience.AlphaPlot.yml:68]:
```suggestion
        tag: "1.011"
```
Nitpick: won't trust yaml that much.

File: io.github.narunlifescience.AlphaPlot.yml (hunk 5 above)
Comment (filipestevao) [reply io.github.narunlifescience.AlphaPlot.yml:16]: The app uses QFileDialog ([reference](https://github.com/narunlifescience/AlphaPlot/blob/70c42625fbe3a3ca81875b27d1ca93efb2955d9d/alphaplot/src/ui/ExtensibleFileDialog.h#L32)).

This is the difference between "Save" and "Load" a project.

AlphaPlot-save-open

Enable "filesystem=home"? What do you suggest?

Comment (bilelmoussaoui) [reply io.github.narunlifescience.AlphaPlot.yml:16]: What I meant is that if the app uses QFileDialog, it uses the portal by default which means you don't need filesystem permissions to open/save files.

Comment (filipestevao) [reply io.github.narunlifescience.AlphaPlot.yml:16]: The image shows that **save** a project does not really require file system permissions, that's correct. However, to **open** a project, it needs filesystem permissions. In **File > Export Graph** opens a similar File Dialog, in which the app needs permissions to export a graph.

Comment (bilelmoussaoui) [reply io.github.narunlifescience.AlphaPlot.yml:16]: Sure, I've figured that the save dialog uses the portal and not the open project, I suppose it's due to not using QFileDialog there?

Comment (filipestevao) [reply io.github.narunlifescience.AlphaPlot.yml:16]: That's right, the dialog for opening a file is different. QFileDialog is not used there.

**Comment by nedrichards** [comment]
bot, build io.github.narunlifescience.AlphaPlot

File: io.github.narunlifescience.AlphaPlot.yml (hunk 5 above)
Comment (barthalion) [reply io.github.narunlifescience.AlphaPlot.yml:16]: I think `--filesystem=home` would be better here, then. Other than that LGTM.

**Pull request closed without merging**